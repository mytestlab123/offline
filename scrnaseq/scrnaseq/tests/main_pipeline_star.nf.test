nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("test-dataset_star_aligner") {

        when {
            // the rest is taken from shared config
            params {
                aligner    = 'star'
                outdir     = "${outputDir}/results_star"

                // Limit resources so that this can run on GitHub Actions -- for some reason it had not been taken from shared config
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {

            assertAll(

                //
                // General assertions
                //

                // Did it finish successfully?
                {assert workflow.success},

                // How many tasks were executed?
                {assert workflow.trace.tasks().size() == 17},

                // How many results were produced?
                {assert path("${outputDir}/results_star").list().size() == 4},
                {assert path("${outputDir}/results_star/star").list().size() == 3},
                {assert path("${outputDir}/results_star/star/mtx_conversions").list().size() == 5},
                {assert path("${outputDir}/results_star/fastqc").list().size() == 12},
                {assert path("${outputDir}/results_star/multiqc").list().size() == 3},

                //
                // Check if files were produced
                //
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_filtered_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.h5ad" ).exists()},

                //
                // Check if files are the same
                //
                {assert snapshot(
                    workflow,
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.SJ.out.tab"                             ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Barcodes.stats"                ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Gene/filtered/matrix.mtx.gz"   ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Gene/filtered/features.tsv.gz" ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Gene/filtered/barcodes.tsv.gz" ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.SJ.out.tab"                             ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Barcodes.stats"                ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Gene/filtered/matrix.mtx.gz"   ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Gene/filtered/features.tsv.gz" ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Gene/filtered/barcodes.tsv.gz" ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_raw_matrix.rds"  ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.rds"  ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_filtered_matrix.rds"  ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.rds"  ),
                ).match()}

            ) // end of assertAll()

        }
    }

}
