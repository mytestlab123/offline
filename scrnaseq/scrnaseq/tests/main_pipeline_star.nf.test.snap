{
    "test-dataset_star_aligner": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [

                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 17,
                    "tasksSucceeded": 17
                },
                "name": "workflow",
                "success": true
            },
            "Sample_X.SJ.out.tab:md5,d2d7f0abe38029012571bdf6622fc6eb",
            "Barcodes.stats:md5,7f99dc8aa5e074fbe5779ea7712c0886",
            "matrix.mtx.gz:md5,6a923393343aa1a69b0cf1bd998c9285",
            "features.tsv.gz:md5,99e453cb1443a3e43e99405184e51a5e",
            "barcodes.tsv.gz:md5,9a7dacaa1779ea43c1507a947fe6992a",
            "Sample_Y.SJ.out.tab:md5,98bd31104a860cf80119dc30d938d163",
            "Barcodes.stats:md5,2dbf1ae426c1afd97903ee001f0db5ce",
            "matrix.mtx.gz:md5,0ae080bd0002e350531a5816e159345e",
            "features.tsv.gz:md5,99e453cb1443a3e43e99405184e51a5e",
            "barcodes.tsv.gz:md5,9b695b0b91bcb146ec9c4688ca10a690",
            "Sample_X_raw_matrix.rds:md5,31604db3e7846acc8d9a60b1a171ce78",
            "Sample_Y_raw_matrix.rds:md5,1a52c823e91acce2b29621c8c99c8c72",
            "Sample_X_filtered_matrix.rds:md5,aa2d36dd8507aba864347c88e4ce0d27",
            "Sample_Y_filtered_matrix.rds:md5,d459af8f99258bcc88b80b2f7c58e911"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-14T16:30:25.7971791"
    }
}
